azurerm/resource_arm_bastion_host_test.go

```diff
@@ -0,0 +1,246 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMBastionHost_basic(t *testing.T) {
+	resourceName := "azurerm_bastion_host.test"
+	ri := tf.AccRandTimeInt()
+	rs := acctest.RandString(4)
+
+	config := testAccAzureRMBastionHost_basic(ri, rs, testLocation())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMBastionHostDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMBastionHostExists(resourceName),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMBastionHost_complete(t *testing.T) {
+	resourceName := "azurerm_bastion_host.test"
+	ri := tf.AccRandTimeInt()
+	rs := acctest.RandString(4)
+
+	config := testAccAzureRMBastionHost_complete(ri, rs, testLocation())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMBastionHostDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMBastionHostExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.environment", "production"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMBastionHost_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	resourceName := "azurerm_bastion_host.test"
+	ri := tf.AccRandTimeInt()
+	rs := acctest.RandString(4)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMBastionHostDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMBastionHost_basic(ri, rs, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMBastionHostExists(resourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMBastionHost_requiresImport(ri, rs, testLocation()),
+				ExpectError: testRequiresImportError("azurerm_bastion_host"),
+			},
+		},
+	})
+}
+
+func testAccAzureRMBastionHost_basic(rInt int, rString string, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
```

Comment:
trying a new thing, would you mind using this here?
```suggestion
  name     = "acctestRG-bastion-%d"
```